### DISCUSSION TITLE
How to get a full command specification

### DISCUSSION THREAD

**Discussion by djrmarques** [pallets/click#2721] [Q&A] [answered]
Hi all! 
So I am looking to create a [carapace specification](https://github.com/carapace-sh/carapace-spec) for a few of the Click cli tools I currently use. Is there anyway I can generate a complete specification in whatever format (JSON for example) of a particular cli I have defined with Click? 
By complete specification, what I mean is a complete description of all the commands and subcommands, as well as related description, types, etc..?

I have looked at the [documentation](https://click.palletsprojects.com/en/8.1.x/shell-completion/#adding-support-for-a-shell) but I really have not been able to figure it out yet.

Thank you :)

**Comment by davidism** [comment, marked as answer]
Objects, starting at the context, have a `to_info_dict` method: https://click.palletsprojects.com/en/8.1.x/api/#click.Context.to_info_dict

**Reply by djrmarques** [reply]
Oh thanks, this really seems to be what I want.
